### PR TITLE
Moving `quad` policies to separate modules

### PR DESCRIPTION
**Pull request by mmahsereci**
# In a Nutshell
This PR moves the existing `quad` policies to separate modules. No functional changes.

# Detailed Description
This is one of the changes proposed in #620 

# Related Issues
Closes #...

### CODE REVIEW

**Comment by mmahsereci** [comment]
@tskarvone I am splitting up the draft PR #620 into smaller chunks, so it' easier to review. This is the first of the set.

File: src/probnum/quad/solvers/policies/_random_policy.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from probnum.quad.solvers.bq_state import BQState
+
+from ._policy import Policy
+
+# pylint: disable=too-few-public-methods, fixme
+
+
+class RandomPolicy(Policy):
+    """Random sampling from an objective.
+    Parameters
```
Comment (JonathanWenger) [inline src/probnum/quad/solvers/policies/_random_policy.py:16]:
```suggestion
    """Random sampling from an objective.
    
    Parameters
```

File: src/probnum/quad/solvers/policies/__init__.py [hunk 2]
```diff
@@ -1 +1,2 @@
-from ._policy import Policy, RandomPolicy
+from ._policy import Policy
```
Comment (JonathanWenger) [inline src/probnum/quad/solvers/policies/__init__.py:1]:
```suggestion
"""Policies / Acquisition functions for Bayesian quadrature."""

from ._policy import Policy
```

**Review by JonathanWenger — APPROVED** [review]
:+1:

File: src/probnum/quad/solvers/policies/__init__.py (hunk 2 above)
Comment (mmahsereci) [reply src/probnum/quad/solvers/policies/__init__.py:3]: changed it to "Policies for Bayesian quadrature." as this module does not contain acquisition functions.

File: src/probnum/quad/solvers/policies/_random_policy.py (hunk 1 above)
Comment (mmahsereci) [reply src/probnum/quad/solvers/policies/_random_policy.py]: done

**Pull request merged**